### ISSUE TITLE
Stop `PLC2701` from flagging underscore-prefixed standard library C modules

### ISSUE THREAD

**Issue by jonathandung** [astral-sh/ruff#24558]
### Summary

Some stdlib modules implemented in C, such as `_collections`, `_functools`, `_heapq` and `_operator`, are determined as private by `PLC2701`, when it is common to import them instead of the pure python counterparts to speed up module startup.

Seeing as though the rule does not trigger for `_pyrepl`, whose name begins with an underscore, the check likely already special-cases some modules, so this change will be as simple as adding the aforementioned to that collection of special cases. Also for consideration are `_asyncio`, which is technically just an accelerator module, and non-C modules like `_collections_abc` and `_weakrefset`.

The following file runs without fault, but ruff flags 52 instances of PLC2701:

```python
import _aix_support
import _android_support
import _apple_support
import _ast
import _ast_unparse
import _asyncio
import _bisect
import _blake2
import _bz2
import _codecs
import _collections
import _compat_pickle
import _contextvars
import _csv
import _ctypes
import _datetime
import _decimal
import _elementtree
import _functools
import _hashlib
import _heapq
import _imp
import _io
import _json
import _locale
import _lzma
import _md5
import _multibytecodec
import _opcode
import _operator
import _pickle
import _queue
import _random
import _sha1
import _sha3
import _signal
import _socket
import _ssl
import _stat
import _statistics
import _string
import _strptime
import _struct
import _symtable
import _testcapi
import _thread
import _threading_local
import _tkinter
import _tracemalloc
import _uuid
import _warnings
import _weakrefset
```

Alternatively, maybe this suppressing behaviour can be opt-in for some cases, or configurable in a list containing allowed private members and modules to import, but that would be more complex.

**Comment by MichaReiser** [comment]
Even if there's a performance benefits, these modules are clearly internals to CPython. I don't think they're documented anywhere. They are also not available in non CPython variants. Adding a configuration option to allow certain imports is something we could consider. How common are these imports in your project? Can't you suppress the import statements with a line or block suppression comment?

**Comment by jonathandung** [comment]
To my knowledge, `_thread` is at least widely used when the higher-level functionality such as `threading.RLock` and `threading.Timer` are not needed. However, I can verify your statement that other implementations may not provide these modules. My project uses these imports in ~15 separate files, and it can grow annoying to noqa them separately, especially when there's no dedicated configuration to suppress them. A pattern, albeit niche, is also used in libraries such as `pytest` and `ruamel.yaml` where all the core logic is put in an underscored-prefixed package (`_pytest`, `_ruamel_yaml`), and it may be desirable to stop this rule from flagging them. Therefore, maybe adding a field corresponding to a list of strings to ignore in the ruff.toml schema is the best way to go.
